data/block/blockV3.go

```diff
@@ -350,10 +345,118 @@ func (hv3 *HeaderV3) CheckFieldsForNil() error {
 	if hv3.SoftwareVersion == nil {
 		return fmt.Errorf("%w in Header.SoftwareVersion", data.ErrNilValue)
 	}
+
+	isGenesisRound := hv3.GetNonce() == 0
+	if isGenesisRound {
+		return nil
+	}
+
+	if hv3.PrevHash == nil {
+		return fmt.Errorf("%w in Header.PrevHash", data.ErrNilValue)
+	}
+	if hv3.PrevRandSeed == nil {
+		return fmt.Errorf("%w in Header.PrevRandSeed", data.ErrNilValue)
+	}
+
+	if hv3.LastExecutionResult == nil {
+		return fmt.Errorf("%w in Header.LastExecutionResult", data.ErrNilValue)
+	}
+
+	return nil
+}
+
+// CheckFieldsIntegrity checks the integrity of the fields
+// It checks a predefined set of fields for nil values or invalid values.
+// It also checks the integrity of LastExecutionResult and ExecutionResults against the header
+func (hv3 *HeaderV3) CheckFieldsIntegrity() error {
```

Comment:
i think we can implement this method for all header types, and add it to the HeaderHandler interface